### PR TITLE
Doc: github default branch

### PR DESCRIPTION
**Pull request by DaedalusG**
Let customers know default branch is configurable on a repos github page, and is respected in default search results

### CODE REVIEW

File: doc/admin/external_service/github.md [hunk 1]
```diff
@@ -129,6 +129,10 @@ GitHub connections support the following configuration options, which are specif
 
 <div markdown-func=jsonschemadoc jsonschemadoc:path="admin/external_service/github.schema.json">[View page on docs.sourcegraph.com](https://docs.sourcegraph.com/admin/external_service/github) to see rendered content.</div>
 
+## Default branch
+
+GitHub uses the master branch as the default branch of a repository. This means that searches targeting a synched github repo will automatically default to results from the master branch. If however you'd like search results to be displayed from another repository by default you may [change a repo's default branch on the github repo settings page](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-branches-in-your-repository/changing-the-default-branch). If this is not an option consider using [search contexts](https://docs.sourcegraph.com/code_search/how-to/search_contexts) instead.
```
Comment (indradhanush) [inline doc/admin/external_service/github.md:134]:
```suggestion
GitHub uses the default branch of a repository for searches and the search results will be shown from the default branch. If you'd like the search results to be displayed from another branch by default, you may [change a repo's default branch on the github repo settings page](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-branches-in-your-repository/changing-the-default-branch). If this is not an option, consider using [search contexts](https://docs.sourcegraph.com/code_search/how-to/search_contexts) instead. 
```

**Review by indradhanush — CHANGES_REQUESTED** [review]
It's not guaranteed that the default branch will be named `master`, so we should just mention `default branch` as the phrase and not use a specific name.

Also left minor suggestions for brevity and punctuation. Thank you for the PR!

File: doc/admin/external_service/github.md (hunk 1 above)
Comment (DaedalusG) [reply doc/admin/external_service/github.md]: Thanks for the clarifications!

File: doc/admin/external_service/github.md [hunk 2]
```diff
@@ -129,6 +129,10 @@ GitHub connections support the following configuration options, which are specif
 
 <div markdown-func=jsonschemadoc jsonschemadoc:path="admin/external_service/github.schema.json">[View page on docs.sourcegraph.com](https://docs.sourcegraph.com/admin/external_service/github) to see rendered content.</div>
 
+## Default branch
+
+Sourcegraph displays search results from the default branch of a repository when no `revision:` parameter is specified. If you'd like the search results to be displayed from another branch by default, you may [change a repo's default branch on the github repo settings page](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-branches-in-your-repository/changing-the-default-branch). If this is not an option, consider using [search contexts](https://docs.sourcegraph.com/code_search/how-to/search_contexts) instead.
```
Comment (Kelvin-Lee) [inline doc/admin/external_service/github.md:134]: Suggest to throw a link to the documentation on the "`revision:`" portion of the first sentence:

https://docs.sourcegraph.com/code_search/reference/queries#repository-revisions

Comment (DaedalusG) [reply doc/admin/external_service/github.md]: Link added ✅

**Review by unknwon — APPROVED** [review]
Could you give this PR a better title?

**Review by indradhanush — APPROVED** [review]

**Review by amberfurbush0317 — COMMENTED** [review]
Looks good Warren!!

**Pull request merged**